ggml-cuda.cu

```diff
@@ -4558,7 +4819,11 @@ static void ggml_mul_mat_q4_0_q8_1_cuda(
     const int compute_capability = g_compute_capabilities[id];
 
     int mmq_x, mmq_y, nwarps;
-    if (compute_capability >= CC_TURING) {
+    if (compute_capability >= CC_OFFSET_AMD) {
```

Comment:
If these tunings don't match with the others set for the mul_mat_q functions earlier then nothing works correctly. Either main freezes or it produces nonsense. Something like this works
```cpp
    if (compute_capability >= CC_RDNA2) {
        mmq_x  =  MMQ_X_Q4_0_RDNA2;
        mmq_y  =  MMQ_Y_Q4_0_RDNA2;
        nwarps = NWARPS_Q4_0_RDNA2;
    } else if (compute_capability >= CC_OFFSET_AMD) {
        mmq_x  =  MMQ_X_Q4_0_RDNA1;
        mmq_y  =  MMQ_Y_Q4_0_RDNA1;
        nwarps = NWARPS_Q4_0_RDNA1;
    } else if (compute_capability >= CC_TURING) {
```
Then a couple of macros near the start of the file
```cpp
#define CC_RDNA2 (CC_OFFSET_AMD + 100*10 + 10*3)
#if defined(__gfx1030__) || defined(__gfx1031__) || defined(__gfx1032__) || defined(__gfx1033__) \
    || defined(__gfx1034__) || defined(__gfx1035__) || defined(__gfx1036__) || defined(__gfx1037__)
#define RDNA2
#endif
```
and the #if for the RDNA2&3 tunings for the mul_mat_q functions could be replaced with `#if defined(RDNA2) || defined(__GFX11__)`

Reply:
Sorry, I forgot to adapt the CPU code. Also what resource are you using to get the GFX values? I didn't know where to look them up so I went by the ones listed in the ROCm documentation supported GPU list.

Reply:
[LLVM doc](https://www.llvm.org/docs/AMDGPUUsage.html#amdgpu-processor-table) should be relatively up-to-date, most of them are not officially supported by ROCm, but people can still build with these architectures.
Alternatively, you can find all of them in LLVM or Mesa code.